### ISSUE TITLE
Train image segmentation with mask or nparray

### ISSUE THREAD

**Issue by Caliphamin** [ultralytics/ultralytics#3000]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

hello everyone
I am working on detect indoor wall and floor using Yolov8-seg
Is there a way to train yolo with mask or nparray instead of text format?

**Comment by Laughing-q** [comment]
@Caliphamin hi, we haven't supported it internally. But I think you can implement it yourself by customizing a new Dataset and Trainer.
Something like this:
```python
from ultralytics.yolo.data import YOLODataset
from ultralytics.yolo.v8 import detect

class CustomDataset(YOLODataset):
    def get_labels():
        """customize how you read the mask and numpy object"""


class CustomTrainer(detect.DetectionTrainer):
    def build_dataset(self, img_path, mode='train', batch=None):
        return CustomDataset(img_path, mode, batch)

settings = dict(data="", epochs=...)
trainer = CustomTrainer(overrides=settings)
trainer.train()
```
